bessctl/conf/testing/run_module_tests.bess

```diff
@@ -0,0 +1,139 @@
+
+    s = socket.socket(socket.AF_UNIX, socket.SOCK_SEQPACKET)
+    s.settimeout(5) #five second timeout
+    s.connect('\0' + SOCKET_PATH + sockname)
+    return socket_port, s
+
+# Craft a packet with the specified IP addresses
+def gen_packet(proto, src_ip, dst_ip):
+    eth = scapy.Ether(src='02:1e:67:9f:4d:ae', dst='06:16:3e:1b:72:32')
+    ip = scapy.IP(src=src_ip, dst=dst_ip)
+    udp = proto(sport=10001, dport=10002)
+    payload = 'helloworld'
+    pkt = eth/ip/udp/payload
+    return str(pkt)
+
+def pkt_str(pkt):
+    if(pkt == None):
+        return "None"
+    else:
+        return pkt.summary()
+
+# Convert an IP address to an integer
+def aton(ip):
+    return socket.inet_aton(ip)
+
+test_packets = [
+  gen_packet(scapy.UDP, '172.16.100.1', '10.0.0.1'),
+  gen_packet(scapy.UDP, '172.12.55.99', '12.34.56.78'),
+  gen_packet(scapy.UDP, '172.12.55.99', '10.0.0.1'),
+  gen_packet(scapy.UDP, '172.16.100.1', '12.34.56.78'),
+  gen_packet(scapy.TCP, '172.12.55.99', '12.34.56.78'),
+  gen_packet(scapy.UDP, '192.168.1.123', '12.34.56.78')
+  ]
+
+## TEST FUNCTIONS ##
+
+# All this does is subject a module to load and make sure it
+# doesn't crash
+def crash_test(module, num_input_ports, num_output_ports):
+    for i in range(num_input_ports):
+        src = Source()
+        src -> Rewrite(templates=test_packets) -> i:module
+    for i in range(num_output_ports):
+        sink = Sink()
+        module:i -> sink
+
+    time.sleep(5)
+
+
+## RUN TEST LOOP ##
+
+for file in glob.glob("conf/testing/module_tests/*.py"):
+    bess.reset_all()
+    CRASH_TEST_INPUTS = []
+    OUTPUT_TEST_INPUTS = []
+    CUSTOM_TEST_FUNCTIONS = []
+
+    print("-- Running Module Tests for " + file + " --")
+    exec(sugar.xform_file(file))
+
+    crash_testid = 1
+    for test in CRASH_TEST_INPUTS:
+        crash_test(test[0], test[1], test[2])
+        print("   " + str(test[0]) + " crash test " + str(crash_testid) + ": PASS")
+        crash_testid += 1
+
+    output_testid = 1
+    for test in OUTPUT_TEST_INPUTS:
+        input_ports = []
+        output_ports = []
+        sockets = []
+
+        for port_num in range(max(test[1],test[2])):
+            sockname = str(output_testid) + "moduletesting-" + SCRIPT_STARTTIME + str(port_num)
+            socket_port, mysocket = gen_socket_and_port(sockname)
+            input_ports.append(PortInc(port=sockname))
+            output_ports.append(PortOut(port=sockname))
+            sockets.append(mysocket)
+            if (port_num < test[1]):
+                input_ports[port_num] -> port_num:test[0]
+            if (port_num < test[2]):
+                test[0]:port_num -> output_ports[port_num]
+
+        bess.resume_all()
+
+        for test_case in test[3]:
+            input_port = test_case["input_port"]
+            output_port = test_case["output_port"]
+            input_pkt = test_case["input_packet"]
+            output_pkt = test_case["output_packet"]
+            if(input_pkt != None):
```

Comment:
It may be worth running pylint:

https://google.github.io/styleguide/pyguide.html#Lint

To catch these inconsistencies in formatting.